src/Uno.Toolkit.UI/Controls/ZoomContentControl/ZoomContentControl.xaml.cs

```diff
@@ -0,0 +1,777 @@
+	}
+
+	private void RegisterPropertyHandlers()
+	{
+		// Register for property changed events.
+		RegisterPropertyChangedCallback(ZoomLevelProperty, OnZoomLevelChanged);
+		RegisterPropertyChangedCallback(MinZoomLevelProperty, CoerceZoomLevel);
+		RegisterPropertyChangedCallback(MaxZoomLevelProperty, CoerceZoomLevel);
+
+		RegisterPropertyChangedCallback(HorizontalOffsetProperty, OnHorizontalOffsetChanged);
+		RegisterPropertyChangedCallback(VerticalOffsetProperty, OnVerticalOffsetChanged);
+
+		RegisterPropertyChangedCallback(IsActiveProperty, IsActiveChanged);
+
+		this.SizeChanged += (s, e) => UpdateContentBoundsVisibility();
+	}
+
+	private void OnVerticalOffsetChanged(DependencyObject sender, DependencyProperty dp)
+	{
+		UpdateContentBoundsVisibility();
+		UpdateScrollVisibility();
+		UpdateVerticalScrollBarValue();
+	}
+
+	private void OnHorizontalOffsetChanged(DependencyObject sender, DependencyProperty dp)
+	{
+		UpdateContentBoundsVisibility();
+		UpdateScrollVisibility();
+		UpdateHorizontalScrollBarValue();
+	}
+
+	private async void OnZoomLevelChanged(DependencyObject sender, DependencyProperty dp)
+	{
+		CoerceZoomLevel(sender, dp);
+		UpdateScrollLimits();
+		UpdateContentBoundsVisibility();
+		UpdateScrollVisibility();
+		await RaiseRenderedContentUpdated();
+	}
+
+	public record BoundsVisibilityData
+	{
+		public bool LeftEdge { get; init; }
+
+		public bool TopEdge { get; init; }
+
+		public bool RightEdge { get; init; }
+
+		public bool BottomEdge { get; init; }
+
+		public bool AllEdgesVisible { get; init; }
+
+		public BoundsVisibilityData(bool leftEdge, bool topEdge, bool rightEdge, bool bottomEdge)
+		{
+			LeftEdge = leftEdge;
+			TopEdge = topEdge;
+			RightEdge = rightEdge;
+			BottomEdge = bottomEdge;
+			AllEdgesVisible = LeftEdge && TopEdge && RightEdge && BottomEdge;
+		}
+	}
+
+	private void UpdateContentBoundsVisibility()
+	{
+		if (_presenter?.Content is FrameworkElement fe)
+		{
+			var m = GetPositionMatrix(fe, this);
+
+			ContentBoundsVisibility = new BoundsVisibilityData(
+				m.OffsetX >= 0,
+				m.OffsetY >= 0,
+				this.ActualWidth >= (fe.ActualWidth * ZoomLevel) + m.OffsetX,
+				this.ActualHeight >= (fe.ActualHeight * ZoomLevel) + m.OffsetY);
+		}
+	}
+
+	private void UpdateScrollVisibility()
+	{
+		IsHorizontalScrollBarVisible = !(ContentBoundsVisibility.RightEdge && ContentBoundsVisibility.LeftEdge);
+		IsVerticalScrollBarVisible = !(ContentBoundsVisibility.TopEdge && ContentBoundsVisibility.BottomEdge);
+	}
+
+	private bool CanMoveThisWay()
+	{
+		if (ContentBoundsVisibility.AllEdgesVisible)
+		{
+			return false;
+		}
+
+		var rules = false;
+		rules |= !ContentBoundsVisibility.LeftEdge && _movimentDirection.Horizontal is true;
+		rules |= !ContentBoundsVisibility.RightEdge && _movimentDirection.Horizontal is false;
+		rules |= !ContentBoundsVisibility.TopEdge && _movimentDirection.Vertical is true;
+		rules |= !ContentBoundsVisibility.BottomEdge && _movimentDirection.Vertical is false;
```

Comment:
not using the helper methods just below?
```suggestion
		rules |= CanScrollLeft() && _movimentDirection.Horizontal is true;
		rules |= CanScrollRight() && _movimentDirection.Horizontal is false;
		rules |= CanScrollUp() && _movimentDirection.Vertical is true;
		rules |= CanScrollDown() && _movimentDirection.Vertical is false;
```